src/main/java/com/example/sample/Application.java

```diff
@@ -33,7 +33,7 @@ String home() {
         }
 
         @GetMapping("/profile")
-        @PreAuthorize("hasAuthority('SCOPE_profile')")
+        @PreAuthorize("hasAuthority('SCOPE_profile') && hasAuthority('SCOPE_offline_access')")
```

Comment:
I'm not sure checking the offline access scope here as a requirement for this call is necessary. Can we please remove this line change?